### PR TITLE
Update react and add field check function

### PR DESCRIPTION
**Pull request by svaldivia**
# Work Done
- Updated to react latest version (16.12)
- Added `areFormFieldsReady()` to be used with the common patterns recommended in the new version of react. [link](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#initializing-state)

#TODO
- Update README

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "babel-plugin-transform-object-assign": "^6.22.0",
     "babel-polyfill": "^6.26.0",
     "babel-preset-env": "^1.6.1",
-    "react": "^15.4.2"
+    "react": "^16.12.0"
```
Comment (davidvandusen) [inline package.json]: What do you think about making this a peer dependency?

File: src/gandalf.ts [hunk 2]
```diff
@@ -68,6 +65,10 @@ class Gandalf extends React.Component<{}, GandalfState> {
       return formValues;
     }, {});
   }
+
+  areFormFieldsReady(): boolean {
```
Comment (davidvandusen) [inline src/gandalf.ts]: They're referred to as "fields" elsewhere, so it's unusual to see them referred to as "form fields" here.

Comment (svaldivia) [reply src/gandalf.ts]: So are you suggesting something like `areFieldsReady()`?

Comment (davidvandusen) [reply src/gandalf.ts]: That would be more consistent, right?

File: package.json (hunk 1 above)
Comment (svaldivia) [reply package.json]: That sounds like a good idea. Would leaving it like `"^16.12.0"` be fine? Then it will support all of react 16

**Pull request merged**